lib/internal/crypto/diffiehellman.js

```diff
@@ -346,6 +346,46 @@ function deriveBitsDH(publicKey, privateKey, callback) {
   job.run();
 }
 
+async function asyncDiffieHellman(publicKey, privateKey) {
+  const { asymmetricKeyType } = privateKey;
+
+  if (asymmetricKeyType === 'dh') {
+    return new Promise((resolve, reject) => {
```

Comment:
You can use `util.promisify` to avoid this `new Promise` dance

Reply:
I believe this to be more efficient than the promisify implementation. Not a hill I want to die on tho.

Reply:
It's exactly as efficient since in recent version this code is what promisify generates - but not a hill to die on for me either was just trying to comment :]

Reply:
https://github.com/nodejs/node/blob/v15.8.0/lib/internal/util.js#L292-L326 sure it has the same outcome, but i think it's handling way more cases which we don't need.